dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -0,0 +1,263 @@
+
+                    progress(channelStatus);
+                }
+                else
+                {
+                    channelRequestResult = channelRequestException.code();
+                    status = PushNotificationChannelStatus::CompletedFailure;
+                    break;
+                }
+            }
+
+            co_await winrt::resume_after(std::chrono::seconds(backOffTimeInSeconds));
+        }
+
+
+        if (status == PushNotificationChannelStatus::CompletedSuccess)
+        {
+            // Returns a com_ptr to the implementation type
+            auto pushChannel =
+                winrt::make_self<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationChannel>(pushChannelReceived);
+
+            channelResult = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationCreateChannelResult>(
+                *(pushChannel.get()),
+                channelRequestResult,
+                status);
+        }
+        else if (status == PushNotificationChannelStatus::CompletedFailure)
+        {
+            channelResult = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationCreateChannelResult>(
+                nullptr,
+                channelRequestResult,
+                status);
+        }
+
+        co_return channelResult;
+    }
+
+    PushNotificationRegistrationToken PushNotificationManager::RegisterActivator(PushNotificationActivationInfo const& details)
+    {
+        winrt::guid taskClsid = details.TaskClsid();
+        DWORD cookie = 0;
+        BackgroundTaskRegistration registeredTask = nullptr;
+
+        if (WI_IsFlagSet(details.Kind(), PushNotificationRegistrationKind::PushTrigger))
+        {
+            bool taskRegistered = false;
+            for (auto task : BackgroundTaskRegistration::AllTasks())
+            {
+                if (task.Value().Name() == backgroundTaskName)
+                {
+                    taskRegistered = true;
+                    break;
+                }
+            }
+
+            if (!taskRegistered)
+            {
+                BackgroundTaskBuilder builder;
+                builder.Name(backgroundTaskName);
+
+                PushNotificationTrigger trigger{};
+                builder.SetTrigger(trigger);
+
+                if (taskClsid != winrt::guid())
+                {
+                    auto builder5 = builder.try_as<winrt::IBackgroundTaskBuilder5>();
+                    if (IsPackagedProcess() && builder5)
+                    {
+                        builder5.SetTaskEntryPointClsid(taskClsid);
+                        winrt::com_array<winrt::IBackgroundCondition> conditions = details.GetConditions();
+                        for (auto condition : conditions)
+                        {
+                            builder.AddCondition(condition);
+                        }
+                    }
```

Comment:
Shouldn't there be some sort of `else throw winrt::hresult_not_implemented()` / `hresult_invalid_argument()`?

Otherwise, a caller is going to pass a `taskClsid`, and we will return success, but in fact their CLSID will never be called.